api/src/services/payload.ts

```diff
@@ -672,31 +672,39 @@ export class PayloadService {
 				if (alterations.create) {
 					const sortField = relation.meta.sort_field;
 
-					const processCreateItem = async (item: Record<string, any>, index: number) => {
-						const record = cloneDeep(item);
-
-						// add sort field value if it is not supplied in the item
-						if (sortField !== null && parent !== null && record[sortField] === undefined) {
-							const highestOrderNumber: Record<string, any> | undefined = await this.knex
-								.select(sortField)
-								.from(relation.collection)
-								.where({ [relation.field]: parent })
-								.whereNotNull(sortField)
-								.orderBy(sortField, 'desc')
-								.first();
-
-							record[sortField] = highestOrderNumber?.[sortField]
-								? highestOrderNumber[sortField] + index + 1
-								: index + 1;
-						}
+					let createPayload: Alterations['create'];
+
+					if (sortField !== null) {
+						const highestOrderNumber: Record<string, any> | undefined = await this.knex
+							.select(sortField)
+							.from(relation.collection)
+							.where({ [relation.field]: parent })
+							.whereNotNull(sortField)
+							.orderBy(sortField, 'desc')
+							.first();
```

Comment:
Lets use `.max()` instead of order by + limit 1 (first). Even though the difference is tiny, it is technically faster 🙂 (and imo also a little easier to read, as it's more explicit)

Reply:
Using `.max()` now 👍